src/components/profile-page/access-tokens/access-token-creation-form.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import type { ChangeEvent, FormEvent } from 'react'
+import React, { Fragment, useCallback, useMemo, useState } from 'react'
+import { Button, Form } from 'react-bootstrap'
+import { cypressId } from '../../../utils/cypress-attribute'
+import { Trans, useTranslation } from 'react-i18next'
+import { AccessTokenCreatedModal } from './access-token-created-modal'
+import { DateTime } from 'luxon'
+import type { AccessTokenWithSecret } from '../../../api/tokens/types'
+import { postNewAccessToken } from '../../../api/tokens'
+import { showErrorNotification } from '../../../redux/ui-notifications/methods'
+
+interface NewTokenFormValues {
+  label: string
+  expiryDate: string
+}
+
+/**
+ * Form for creating a new access token.
+ */
+export const AccessTokenCreationForm: React.FC = () => {
```

Comment:
:banana: 
This file is very long :/ 140 LOC

The form controls (and the following button) are good candidates for component extractions. Even if the resulting components don't contain any processing logic, it would improve the readablity of this component because it reduces the loc in this file. You could also extract stuff like onCreateToken into custom hooks.

Reply:
129 code lines, but yes - I'm changing that.

![image](https://user-images.githubusercontent.com/52606896/143770338-10900237-045d-4f1e-82ca-088c8fe82188.png)